circuits/cpp/barretenberg/cpp/src/barretenberg/dsl/acir_format/fixed_base_scalar_mul.cpp

```diff
@@ -1,17 +1,39 @@
 #include "fixed_base_scalar_mul.hpp"
+#include "barretenberg/dsl/types.hpp"
+#include "barretenberg/ecc/curves/bn254/fr.hpp"
+#include "barretenberg/ecc/curves/grumpkin/grumpkin.hpp"
+#include "barretenberg/proof_system/arithmetization/gate_data.hpp"
 
 namespace acir_format {
 
 void create_fixed_base_constraint(Builder& builder, const FixedBaseScalarMul& input)
 {
 
+    // Computes low * G + high * 2^128 * G
+    //
+    // Low and high need to be less than 2^128
     field_ct low_as_field = field_ct::from_witness_index(&builder, input.low);
     field_ct high_as_field = field_ct::from_witness_index(&builder, input.high);
-    (void)high_as_field;
-    auto public_key = group_ct::fixed_base_scalar_mul_g1<254>(low_as_field);
 
-    builder.assert_equal(public_key.x.witness_index, input.pub_key_x);
-    builder.assert_equal(public_key.y.witness_index, input.pub_key_y);
+    auto low_value = grumpkin::fr(low_as_field.get_value());
+    auto high_value = grumpkin::fr(high_as_field.get_value());
+    auto pow_128 = grumpkin::fr(2).pow(128);
+
+    grumpkin::g1::element result = grumpkin::g1::one * low_value + grumpkin::g1::one * (high_value * pow_128);
+    grumpkin::g1::affine_element result_affine = result.normalize();
+
+    auto x_var = builder.add_variable(result_affine.x);
+    auto y_var = builder.add_variable(result_affine.y);
+    builder.create_add_gate({ x_var,
```

Comment:
Added this dummy gate so that the x_var and y_var have a gate constrained to them, IIRC this was an implicit assumption by the barretenberg library